crates/bevy_render/src/camera/camera.rs

```diff
@@ -219,6 +225,17 @@ impl Camera {
     ///
     /// To get the coordinates in Normalized Device Coordinates, you should use
     /// [`world_to_ndc`](Self::world_to_ndc).
+    ///
+    /// Returns `None` if any of these conditions occur:
+    /// - The logical viewport size cannot be computed. This can happen if both:
+    ///   - The viewport is not set
+    ///   - Bevy has not set the render target. This can happen if:
+    ///     - The function is called just after the `Camera` is created, before `camera_system` is executed
+    ///     - The target isn't correctly set (for example it points to an image that doesn't exist)
+    /// - The projection matrix is invalid
+    /// - The camera transform is invalid or cannot be inverted
+    /// - The world position is invalid
+    /// - The computed coordinates are beyond the near or far plane
```

Comment:
On this last point:

I think this reason should be the first in the least, it's the only "unexpected" one, the others are all like "something went wrong", this one is a design choice (to exclude stuff outside the frustum but only for front and back, not left/right or up/down). I think it kinda makes sense otherwise since we discard the z component we have no way to know if it's in the frustum or not, but it should be more "advertised".

Also, ***technically*** I feel like despite was is said in the comment bellow, and this is the case by default, if the "implicit frustum" defined by the projection matrix (the one used to actually render stuff), and the "actual frustum" (the one used to exclude stuff from the rendering process), are not the same (which, I think, can happen, but you'd have to do some weird stuff), this uses the "implicit frustum", not the actual one, that maybe should be mentioned?

Reply:
I don't really understand this, but I've added a parentheses note

Reply:
I think you can change the line to "The computed coordinates are beyond the near or far plane defined by the projection,". Or maybe link to https://docs.rs/bevy/latest/bevy/render/camera/enum.Projection.html:

```rust
The computed coordinates are beyond the near or far plane defined by the [`Projection`],
```

I would also change the comment inside the code bellow to "(implicit, defined by the projection)"

For more details if you're interested, if you look at, for example, [Camera3dBundle](https://docs.rs/bevy/latest/bevy/core_pipeline/core_3d/struct.Camera3dBundle.html), you'll notice it has a [Projection](https://docs.rs/bevy/latest/bevy/render/camera/enum.Projection.html) component, and a [Frustum](https://docs.rs/bevy/latest/bevy/render/primitives/struct.Frustum.html) component.

The projection is used to transform all the objects of the world to fit them in a small box in front of the camera, this box will be "squished" to a rectangle that will make up the image rendered by the camera.

But because everything that won't end up in that box don't need to be transformed, we optimize beforehand which ones need to be. For there is the frustum, which is a kind of truncated pyramid that defines the limits of what the camera can see. We can check if an object is in the pyramid, if it isn't we don't consider it for rendering at all.

In almost all scenarios, the Frustum is calculated from the Projection, the truncated pyramid will become the rendering box after projection, so the near and far plane of both (where the pyramid is truncated for the Frustum, and the limits of the projection box) are the same. But in some specific scenarios they can not be, hence why I think it's important to specify it explicitly here.